sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -325,6 +325,21 @@ class SessionCatalog(
     new Path(new Path(dbLocation), formatTableName(tableIdent.table)).toString
   }
 
+  // ----------------------------------------------
+  // | Methods that interact with temp views only |
+  // ----------------------------------------------
+
+  /**
+   * Return a temporary view exactly as it was stored.
+   */
+  def getTempView(name: String): Option[LogicalPlan] = synchronized {
```

Comment:
In the same file, there are also interfaces like isTemporaryTable, tableExists, refreshTable, clearTempTables, which can accept a temporary view, but using the name "table" instead of "view"

I am not sure introducing getTempView and dropTempView will make the class SessionCatalog's interface simpler or more complicated.